### DISCUSSION TITLE
Unable to ping across Site-to-Site VPN but tunnel connection is up

### DISCUSSION THREAD

**Discussion by Nick-zaw** [strongswan/strongswan#2541] [Q&A]
Hello,
I am very new to this, so please be easy on me. When providing instructions, it would be very helpful if you could provide the full command I need to run.

We are trying to set up a site-to-site VPN connection between Azure WAN and a Strongswan gateway hosted in one of our data centers. On the Strongswan gateway side, we need to configure DNAT and SNAT due to a subnet conflict with our other data center. The setup we're aiming for is Azure WAN acting as the hub, with multiple sites connecting to Azure WAN as spokes. Additionally, we want to establish branch-to-branch connections.

When we ping the subnet behind the Strongswan gateway, we are seeing a "Request Timeout." We performed a packet capture on the Strongswan gateway and observed that the ICMP packets are reaching the gateway. I suspect we are missing some configuration on the Strongswan gateway related to NAT or iptables rules.

Any help would be greatly appreciated.
[Strongswan_requested_logs.txt](https://github.com/user-attachments/files/17794579/Strongswan_requested_logs.txt)

Local peer ip : 172.21.179.205
Local subnet - 172.21.179.0/24
Nat subnet - 6.11.3.0/24

Remote peer ip - 57.155.55.113
Remote subnets - 6.11.1.0/24

Thank you and have a great day

**Comment by tobiasbrunner** [comment]
> we need to configure DNAT and SNAT due to a subnet conflict with our other data center

Hm, wouldn't you maybe want to use `NETMAP` instead (see [this example](https://strongswan.org/testing/testresults/ikev2/net2net-same-nets/))?

Also, your config uses marks. So only traffic with that mark applied gets tunneled.

**Reply by Nick-zaw** [reply]
Hello @tobiasbrunner , 

Thank you for your reply. Would you mind providing an example of how we can mark the traffic using a route-based VPN? We would need one to one static nat for this connections.

below is the output of ip xfrm policy and ip xfrm state

# ip xfrm policy
src 0.0.0.0/0 dst 0.0.0.0/0
        dir out priority 399999
        mark 0x2a/0xffffffff
        tmpl src 172.21.179.205 dst 57.155.55.113
                proto esp spi 0x3e51d57a reqid 1 mode tunnel
src 0.0.0.0/0 dst 0.0.0.0/0
        dir fwd priority 399999
        mark 0x2a/0xffffffff
        tmpl src 57.155.55.113 dst 172.21.179.205
                proto esp reqid 1 mode tunnel
src 0.0.0.0/0 dst 0.0.0.0/0
        dir in priority 399999
        mark 0x2a/0xffffffff
        tmpl src 57.155.55.113 dst 172.21.179.205
                proto esp reqid 1 mode tunnel
src 0.0.0.0/0 dst 0.0.0.0/0
        socket in priority 0
src 0.0.0.0/0 dst 0.0.0.0/0
        socket out priority 0
src 0.0.0.0/0 dst 0.0.0.0/0
        socket in priority 0
src 0.0.0.0/0 dst 0.0.0.0/0
        socket out priority 0
src ::/0 dst ::/0
        socket in priority 0
src ::/0 dst ::/0
        socket out priority 0
src ::/0 dst ::/0
        socket in priority 0
src ::/0 dst ::/0
        socket out priority 0
# ip xfrm state
src 172.21.179.205 dst 57.155.55.113
        proto esp spi 0x3e51d57a reqid 1 mode tunnel
        replay-window 0 flag af-unspec
        mark 0x2a/0xffffffff
        auth-trunc hmac(sha256) 0x5f82f92214213286cd97df0544e73c7ce10bf08ef1554a5ae11a1c07444ccba7 128
        enc cbc(aes) 0x12c78c75e343c16e2dc133739b7e182777244c041f4afa5236bbec467ab8aa6e
        encap type espinudp sport 4500 dport 4500 addr 0.0.0.0
        anti-replay context: seq 0x0, oseq 0x0, bitmap 0x00000000
src 57.155.55.113 dst 172.21.179.205
        proto esp spi 0xc43d50b6 reqid 1 mode tunnel
        replay-window 32 flag af-unspec
        auth-trunc hmac(sha256) 0xd14c8dac08b40125e9f3f8e7d87a72232c404676b490959e3d5cc61a978e9b8d 128
        enc cbc(aes) 0x934fcb624b366e8733bd3c72c417cc8572d2ab24fcb284b1077ccd522d81a1c8
        encap type espinudp sport 4500 dport 4500 addr 0.0.0.0
        anti-replay context: seq 0x0, oseq 0x0, bitmap 0x00000000

**Reply by tobiasbrunner** [reply]
Please refer to [the docs](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html).

**Reply by Nick-zaw** [reply]
I carefully followed the steps in the documentation to mark the traffic, but I am still unable to ping the server behind the Strongswan gateway. We would greatly appreciate it if you could point out any misconfiguration.

Thank you and have a great day

iptables-save
=========
# Generated by iptables-save v1.8.7 on Tue Nov 19 07:40:09 2024
*mangle
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
-A PREROUTING -s 6.11.1.10/32 -d 6.11.3.10/32 -i ipsec0 -j MARK --set-xmark 0x2a/0xffffffff
COMMIT
# Completed on Tue Nov 19 07:40:09 2024
# Generated by iptables-save v1.8.7 on Tue Nov 19 07:40:09 2024
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
COMMIT
# Completed on Tue Nov 19 07:40:09 2024
# Generated by iptables-save v1.8.7 on Tue Nov 19 07:40:09 2024
*nat
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
-A PREROUTING -d 6.11.3.10/32 -i ipsec0 -m mark --mark 0x2a -j DNAT --to-destination 172.21.179.10
COMMIT
# Completed on Tue Nov 19 07:40:09 2024


ip -s xfrm policy
===========
src 0.0.0.0/0 dst 0.0.0.0/0 uid 0
        dir out action allow index 873 priority 399999 share any flag  (0x00000000)
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 0(sec), hard 0(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          0(bytes), 0(packets)
          add 2024-11-19 07:39:08 use -
        mark 0x2a/0xffffffff
        tmpl src 172.21.179.205 dst 57.155.55.113
                proto esp spi 0x31d9bccc(836353228) reqid 1(0x00000001) mode tunnel
                level required share any
                enc-mask ffffffff auth-mask ffffffff comp-mask ffffffff
src 0.0.0.0/0 dst 0.0.0.0/0 uid 0
        dir fwd action allow index 866 priority 399999 share any flag  (0x00000000)
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 0(sec), hard 0(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          0(bytes), 0(packets)
          add 2024-11-19 07:39:08 use -
        mark 0x2a/0xffffffff
        tmpl src 57.155.55.113 dst 172.21.179.205
                proto esp spi 0x00000000(0) reqid 1(0x00000001) mode tunnel
                level required share any
                enc-mask ffffffff auth-mask ffffffff comp-mask ffffffff
src 0.0.0.0/0 dst 0.0.0.0/0 uid 0
        dir in action allow index 856 priority 399999 share any flag  (0x00000000)
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 0(sec), hard 0(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          0(bytes), 0(packets)
          add 2024-11-19 07:39:08 use 2024-11-19 07:40:45
        mark 0x2a/0xffffffff
        tmpl src 57.155.55.113 dst 172.21.179.205
                proto esp spi 0x00000000(0) reqid 1(0x00000001) mode tunnel
                level required share any
                enc-mask ffffffff auth-mask ffffffff comp-mask ffffffff

ip -s xfrm state
==========
src 172.21.179.205 dst 57.155.55.113
        proto esp spi 0x31d9bccc(836353228) reqid 1(0x00000001) mode tunnel
        replay-window 0 seq 0x00000000 flag af-unspec (0x00100000)
        mark 0x2a/0xffffffff
        auth-trunc hmac(sha256) 0x0609e8fa3711afa73df57a256bae4e67429e1c70611b187e02affd1ba5aab8f3 (256 bits) 128
        enc cbc(aes) 0xa1ed81dc2fcfad29e7b49363d8684221eb36f4176972420e5dc1427ae4dd492d (256 bits)
        encap type espinudp sport 4500 dport 4500 addr 0.0.0.0
        anti-replay context: seq 0x0, oseq 0x0, bitmap 0x00000000
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 3353(sec), hard 3960(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          0(bytes), 0(packets)
          add 2024-11-19 07:39:08 use -
        stats:
          replay-window 0 replay 0 failed 0
src 57.155.55.113 dst 172.21.179.205
        proto esp spi 0xc91e1830(3374192688) reqid 1(0x00000001) mode tunnel
        replay-window 32 seq 0x00000000 flag af-unspec (0x00100000)
        auth-trunc hmac(sha256) 0x5497f99467a896ebf0520a3618f905bef69a1ffdfb8a9e363bef73c484c07f9f (256 bits) 128
        enc cbc(aes) 0x05e01edbb26a4b77e4798a6db9b275176f80f9f5b398cbe0c0bdde37a8bd9fed (256 bits)
        encap type espinudp sport 4500 dport 4500 addr 0.0.0.0
        anti-replay context: seq 0x5, oseq 0x0, bitmap 0x0000001f
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 3558(sec), hard 3960(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          300(bytes), 5(packets)
          add 2024-11-19 07:39:08 use 2024-11-19 07:39:35
        stats:
          replay-window 0 replay 0 failed 0

**Reply by tobiasbrunner** [reply]
What's that `ipsec0` interface you refer to in your rules? I also don't see an `SNAT` rule. And you don't need to match the mark there either (in particular for the `DNAT` rule as the inbound packets won't have the mark unless you specifically add it manually or via `set_mark_in`).

**Reply by Nick-zaw** [reply]
Thank you for your reply. The ipsec0 interface in the rules is the VTI interface I created for the tunnel. I have amended the iptables rules as you suggested, but I still do not see the ICMP reply from the end host. We would greatly appreciate if you could point out any misconfiguration.

Thank you and have a great day.


 iptables-save
=========
# Generated by iptables-save v1.8.7 on Tue Nov 19 08:55:22 2024
*mangle
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
-A PREROUTING -s 6.11.1.10/32 -d 6.11.3.10/32 -i ipsec0 -j MARK --set-xmark 0x2a/0xffffffff
COMMIT
# Completed on Tue Nov 19 08:55:22 2024
# Generated by iptables-save v1.8.7 on Tue Nov 19 08:55:22 2024
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
COMMIT
# Completed on Tue Nov 19 08:55:22 2024
# Generated by iptables-save v1.8.7 on Tue Nov 19 08:55:22 2024
*nat
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
-A PREROUTING -d 6.11.3.10/32 -j DNAT --to-destination 172.21.179.10
-A POSTROUTING -s 172.21.179.10/32 -j SNAT --to-source 6.11.3.10
COMMIT
# Completed on Tue Nov 19 08:55:22 2024

tcpdump -nn -i any host 6.11.1.10
======================
tcpdump: data link type LINUX_SLL2
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on any, link-type LINUX_SLL2 (Linux cooked v2), snapshot length 262144 bytes
08:55:45.213306 ipsec0 In  IP 6.11.1.10 > 6.11.3.10: ICMP echo request, id 37796, seq 8805, length 40
08:55:50.211404 ipsec0 In  IP 6.11.1.10 > 6.11.3.10: ICMP echo request, id 50410, seq 8806, length 40
08:55:55.199963 ipsec0 In  IP 6.11.1.10 > 6.11.3.10: ICMP echo request, id 62402, seq 8807, length 40
^C
3 packets captured
4 packets received by filter
0 packets dropped by kernel

ifconfig
=====
ens160: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.21.179.205  netmask 255.255.255.0  broadcast 172.21.179.255
        inet6 fe80::250:56ff:fe9e:8db3  prefixlen 64  scopeid 0x20<link>
        ether 00:50:56:9e:8d:b3  txqueuelen 1000  (Ethernet)
        RX packets 197957  bytes 77289273 (77.2 MB)
        RX errors 0  dropped 73  overruns 0  frame 0
        TX packets 109995  bytes 15973368 (15.9 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ipsec0: flags=209<UP,POINTOPOINT,RUNNING,NOARP>  mtu 1480
        inet 6.11.3.12  netmask 255.255.255.252  destination 6.11.3.12
        inet6 fe80::5efe:ac15:b3cd  prefixlen 64  scopeid 0x20<link>
        tunnel   txqueuelen 1000  (IPIP Tunnel)
        RX packets 3059  bytes 188716 (188.7 KB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 690  bytes 47160 (47.1 KB)
        TX errors 5  dropped 0 overruns 0  carrier 5  collisions 0
